lib/node/validator.go

```diff
@@ -0,0 +1,209 @@
+package sebaknode
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"sync"
+
+	"boscoin.io/sebak/lib/common"
+
+	"github.com/stellar/go/keypair"
+)
+
+type LocalNodeFromJSON struct {
+	Alias    string                `json:"alias"`
+	Address  string                `json:"address"`
+	Endpoint *sebakcommon.Endpoint `json:"endpoint"`
+	State    NodeState             `json:"state"`
+}
+
+type Validator struct {
+	sync.Mutex
+
+	keypair *keypair.Full
+
+	state    NodeState
+	alias    string
+	address  string
+	endpoint *sebakcommon.Endpoint
+}
+
+func (v *Validator) String() string {
+	return v.Alias()
+}
+
+func (v *Validator) Equal(a Node) bool {
+	if v.Address() == a.Address() {
+		return true
+	}
+
+	return false
+}
+
+func (v *Validator) DeepEqual(a Node) bool {
+	if !v.Equal(a) {
+		return false
+	}
+	if v.Endpoint().String() != a.Endpoint().String() {
+		return false
+	}
+
+	return true
+}
+
+func (v *Validator) State() NodeState {
+	return v.state
+}
+
+func (v *Validator) SetBooting() {
+	v.state = NodeStateBOOTING
+}
+
+func (v *Validator) SetCatchup() {
+	v.state = NodeStateCATCHUP
+}
+
+func (v *Validator) SetConsensus() {
+	v.state = NodeStateCONSENSUS
+}
+
+func (v *Validator) SetTerminating() {
+	v.state = NodeStateTERMINATING
+}
+
+func (v *Validator) Address() string {
+	return v.address
+}
+
+func (v *Validator) Keypair() *keypair.Full {
+	return v.keypair
+}
+
+func (v *Validator) SetKeypair(kp *keypair.Full) {
+	v.address = kp.Address()
+	v.keypair = kp
+}
```

Comment:
I guess the `LocalNode` does not know the secret seed(`*keypair.Full`) of it's `Validator`s, so if possible, it would be better that `SetKeypair()` will be empty method, or removed from `Node` interface.